exercises/practice/ocr-numbers/.meta/Example.roc

```diff
@@ -0,0 +1,70 @@
+module [convert]
+
+BadGridSize : [HeightWasNotAMultipleOf4, WidthWasNotAMultipleOf3, GridShapeWasNotRectangular]
+
+convert : Str -> Result Str BadGridSize
+convert = \grid ->
+    if grid == "" then
+        Ok ""
+        else
+
+    gridChars = grid |> Str.split "\n" |> List.map Str.toUtf8
+    size = checkSize? gridChars
+    gridChars
+    |> List.chunksOf 4 # split vertically into groups of 4 rows
+    |> List.map \rowGroup ->
+        getDigitGrids rowGroup size.width
+        |> List.map identifyDigit
+        |> Str.joinWith ""
+    |> Str.joinWith ","
+    |> Ok
+
+checkSize : List (List U8) -> Result { height : U64, width : U64 } BadGridSize
+checkSize = \gridChars ->
+    height = List.len gridChars
+    if height % 4 != 0 then
+        Err HeightWasNotAMultipleOf4
+    else
+        width = gridChars |> List.map List.len |> List.max |> Result.withDefault 0
+        if width % 3 != 0 then
+            Err WidthWasNotAMultipleOf3
+        else
+            isRectangular = gridChars |> List.all \row -> List.len row == width
+            if isRectangular then
+                Ok { height, width }
+            else
+                Err GridShapeWasNotRectangular
```

Comment:
Not super important, but this code seems like it would benefit from the early return if syntax.

Reply:
Good idea, thanks!